internal/proxy/publish_handler.go

```diff
@@ -67,8 +65,11 @@ func (h *PublishHandler) Handle(node *centrifuge.Node) PublishHandlerFunc {
 		publishRep, err := h.config.Proxy.ProxyPublish(client.Context(), req)
 		duration := time.Since(started).Seconds()
 		if err != nil {
-			if errors.Is(err, context.Canceled) {
+			select {
+			case <-client.Context().Done():
+				// Client connection already closed.
 				return centrifuge.PublishReply{}, nil
```

Comment:
Let's return `centrifuge.PublishReply{}, centrifuge.DisconnectNormal` in this case here and in all other proxy types.

Reply:
Fixed in all proxy handlers